zeeguu/core/definition_of_learned/is_learned.py

```diff
@@ -21,14 +24,28 @@ def is_learned_based_on_exercise_outcomes(exercise_log, is_productive=True):
     if exercise_log.is_empty():
         return False
 
-    # For a bookmark to be learned it needs to have 1 or 2 cycles
-    # completed depending on the user preference.
     if exercise_log.last_exercise().is_too_easy():
         return True
 
-    streak_counts = exercise_log.count_number_of_streaks()
-    full_cycles_completed = streak_counts.get(LEARNING_CYCLE_LENGTH)
+    # For a bookmark to be learned it needs to have 1 or 2 cycles
+    # completed depending on the user preference.
+
+    scheduler = get_scheduler(exercise_log.bookmark.user)
+
+    learning_cycle_length = len(scheduler.get_cooling_interval_dictionary())
+
+    streaks_by_length = exercise_log.exercise_streaks_of_given_length()
+    full_cycles_completed = streaks_by_length.get(learning_cycle_length)
+
     if is_productive:
         return full_cycles_completed == 2
     else:
         return full_cycles_completed == 1
```

Comment:
i've been trying to understand this function @tfnribeiro, and in the process also renamed and reformatted a bit the `exercise_streaks_of_given_length`. However, I don't fully understand it. Are we sure it's not buggy? 

In particular this conditional here, don't we need a full_cycles_completed to be 4 for the `Levels` scheduler? Or do I mis-understand what full_cyucles_completed means?